### ISSUE TITLE
UP008 false negative in inner (nested) classes

### ISSUE THREAD

**Issue by generalmimon** [astral-sh/ruff#22597]
### Summary

Playground link: https://play.ruff.rs/f9f8f76b-c468-4f4f-a3b8-0bca5850daf1

```python
class Base:
    def __init__(self, foo):
        self.foo = foo

class Outer(Base):
    def __init__(self, foo):
        super(Outer, self).__init__(foo)  # UP008 issued as expected

    class Inner(Base):
        def __init__(self, foo):
            super(Outer.Inner, self).__init__(foo)  # Bug: would also expect UP008, but it was not issued

o = Outer(5)
i = Outer.Inner(5)
```

---

By the way, Pylint's [`super-with-arguments`](https://pylint.readthedocs.io/en/latest/user_guide/messages/refactor/super-with-arguments.html) also has a problem only with `super(Outer, self)`, but not with `super(Outer.Inner, self)`:

```console
$ pylint --version
pylint 4.0.4
astroid 4.0.3
Python 3.14.2 (tags/v3.14.2:df79316, Dec  5 2025, 17:18:21) [MSC v.1944 64 bit (AMD64)]
$ pylint --disable=all --enable=super-with-arguments test.py
************* Module test
test.py:7:8: R1725: Consider using Python 3 style super() without arguments (super-with-arguments)

-----------------------------------
Your code has been rated at 9.09/10
```

However, https://github.com/asottile/pyupgrade correctly upgrades both `super()` calls:

```console
$ pip freeze | grep -F pyupgrade
pyupgrade==3.21.2
$ cp test.py test-original.py
$ pyupgrade test.py
Rewriting test.py
$ git diff --no-index test-original.py test.py
diff --git 1/test-original.py 2/test.py
index 2f6d7cd2..65bf24ac 100644
--- 1/test-original.py
+++ 2/test.py
@@ -4,11 +4,11 @@ class Base:
 
 class Outer(Base):
     def __init__(self, foo):
-        super(Outer, self).__init__(foo)  # UP008 issued as expected
+        super().__init__(foo)  # UP008 issued as expected
 
     class Inner(Base):
         def __init__(self, foo):
-            super(Outer.Inner, self).__init__(foo)  # Bug: would also expect UP008, but it was not issued
+            super().__init__(foo)  # Bug: would also expect UP008, but it was not issued
 
 o = Outer(5)
 i = Outer.Inner(5)
```

### Version

ruff 0.14.11

**Comment by ntBre** [comment]
Thanks for the report and the great write-up! This does seem like a bug.

**Comment by leandrobbraga** [comment]
I have a fix in https://github.com/astral-sh/ruff/pull/22677

**Comment by generalmimon** [comment]
I went digging in the https://github.com/asottile/pyupgrade repo and found that it also didn't originally support `super` calls in nested classes: https://github.com/asottile/pyupgrade/issues/696

The reason I mention this is that the associated PR https://github.com/asottile/pyupgrade/pull/707 could be interesting to look at, both for the implementation and for the added test cases, which are quite thorough and Ruff would not pass them at the moment. [pyupgrade](https://github.com/asottile/pyupgrade) is MIT-licensed (see [LICENSE](https://github.com/asottile/pyupgrade/blob/369aea6489e38ca74a18df494f191857ba484950/LICENSE)), so I believe it should be OK to look at it.

As for the test cases that Ruff wouldn't pass, this test case https://github.com/asottile/pyupgrade/blob/369aea6489e38ca74a18df494f191857ba484950/tests/features/super_test.py#L32-L35 essentially covers the same situation that I described in my review comment https://github.com/astral-sh/ruff/pull/22677#discussion_r2714271448.
